EngineLayer/Proteomics/CompactPeptide.cs

```diff
@@ -30,6 +32,103 @@ public CompactPeptide(PeptideWithSetModifications peptideWithSetModifications)
             MonoisotopicMassIncludingFixedMods = peptideWithSetModifications.MonoisotopicMass;
         }
 
+        public CompactPeptide(PeptideWithSetModifications peptideWithSetModifications, TerminusType terminusType)
+        {
+            double theMass = 0;
+            NTerminalMasses = new double[1];
+            CTerminalMasses = new double[1];
+            if (terminusType==TerminusType.None || terminusType == TerminusType.N)
+            {
+                if (peptideWithSetModifications.allModsOneIsNterminus.TryGetValue(1, out ModificationWithMass pep_n_term_variable_mod))
+                    foreach (double nl in pep_n_term_variable_mod.neutralLosses)
+                        theMass = pep_n_term_variable_mod.monoisotopicMass - nl;
+                else
+                    theMass = 0;
+                NTerminalMasses = ComputeFollowingFragmentMasses(peptideWithSetModifications, theMass, 1, 1).ToArray();
+            }
+            if (terminusType == TerminusType.None || terminusType == TerminusType.C)
+            {
+                theMass = 0;
+                if (peptideWithSetModifications.allModsOneIsNterminus.TryGetValue(peptideWithSetModifications.Length + 2, out ModificationWithMass pep_c_term_variable_mod))
+                    foreach (double nl in pep_c_term_variable_mod.neutralLosses)
+                        theMass = pep_c_term_variable_mod.monoisotopicMass - nl;
+                else
+                    theMass = 0;
+                CTerminalMasses = ComputeFollowingFragmentMasses(peptideWithSetModifications, theMass, peptideWithSetModifications.Length, -1).ToArray();
+            }
+            MonoisotopicMassIncludingFixedMods = peptideWithSetModifications.MonoisotopicMass;
+        }
+
         #endregion Public Constructors
+
+        #region Public Methods
+
+        public double[] ProductMassesMightHaveDuplicatesAndNaNs(List<ProductType> productTypes)
```

Comment:
I'll move it back to base, it was just never called from the base class or from CPWM